backend/src/v2/compiler/argocompiler/container.go

```diff
@@ -339,6 +343,46 @@ func (c *workflowCompiler) addContainerExecutorTemplate() string {
 			Env:     commonEnvs,
 		},
 	}
+	caBundleCfgMapName := os.Getenv("ARTIFACT_COPY_STEP_CABUNDLE_CONFIGMAP_NAME")
+	caBundleCfgMapKey := os.Getenv("ARTIFACT_COPY_STEP_CABUNDLE_CONFIGMAP_KEY")
+	caBundleMountPath := os.Getenv("ARTIFACT_COPY_STEP_CABUNDLE_MOUNTPATH")
+	if caBundleCfgMapName != "" && caBundleCfgMapKey != "" {
+		var certDirectories = []string{
+			caBundleMountPath,
+			"/etc/ssl/certs",
+			"/etc/pki/tls/certs",
+		}
+		sslCertDir := strings.Join(certDirectories, ":")
+		commonEnvs = append(commonEnvs, k8score.EnvVar{
+			Name:  "SSL_CERT_DIR",
+			Value: sslCertDir,
+		})
+		executor.Container.Env = append(executor.Container.Env, k8score.EnvVar{
+			Name:  "SSL_CERT_DIR",
+			Value: sslCertDir,
+		})
```

Comment:
seems like when I run a pipeline, this: 
![image](https://github.com/opendatahub-io/data-science-pipelines/assets/10904967/cc4c77fa-583c-4b4a-b92b-82d9439b9126)
seems to happen, where the env var is added multiple times

Reply:
have you noticed this behavior as well?

Reply:
it's also inconsistent how many are added: 
![image](https://github.com/opendatahub-io/data-science-pipelines/assets/10904967/f0ad3076-06b0-41c7-9228-4b3e937286ca)

Reply:
my guess you are appending the env field multiple times: 


```golang
commonEnvs = append(commonEnvs, k8score.EnvVar{
			Name:  "SSL_CERT_DIR",
			Value: sslCertDir,
})
```

I'm not sure if this is useful because `commonEnvs` is used [here](https://github.com/opendatahub-io/data-science-pipelines/blob/master/backend/src/v2/compiler/argocompiler/container.go#L339) which is before you update env. But it's a reference type, so it does get included in the env field, however you add it again here: 

```golang
executor.Container.Env = append(executor.Container.Env, k8score.EnvVar{
	Name:  "SSL_CERT_DIR",
	Value: sslCertDir,
})
```

maybe try removing the update to `commonenv` and only keep the last change.

Reply:
> have you noticed this behavior as well?

@HumairAK I will take a look at it. I didn't notice this behaviour when I deployed in my cluster.

Reply:
@HumairAK I am not seeing it when I deployed it in my cluster. But, its working even after removing commonEnvs part. So, I removed them.Please check.